GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -3217,15 +3217,20 @@ public void PopulateFavouriteRepositoriesMenu(ToolStripDropDownItem container, I
 
         private void FormBrowse_DragDrop(object sender, DragEventArgs e)
         {
-            CommitInfoTabControl.SelectedTab = TreeTabPage;
             HandleDrop(e);
         }
 
         private void HandleDrop(DragEventArgs e)
         {
+            if (TreeTabPage.Parent == null)
+            {
+                return;
+            }
+
             var itemPath = (e.Data.GetData(DataFormats.Text) ?? e.Data.GetData(DataFormats.UnicodeText)) as string;
             if (itemPath != null && (File.Exists(itemPath) || Directory.Exists(itemPath)))
```

Comment:
This check should probably be changed so it checks that the file is in this repo. Right now it switches to the Tree tab if the file exist in another repo

Reply:
Check the return value of fileTree.SelectFileOrFolder and change the SelectedTab if the return is true.